### PR TITLE
handled metric for recovery

### PR DESCRIPTION
**Pull request by Laeeqdev**
feat [4952](https://github.com/devtron-labs/devtron/issues/4952)

### CODE REVIEW

File: pubsub-lib/metrics/metrics.go [hunk 1]
```diff
@@ -32,6 +32,10 @@ var NatsEventDeliveryCount = promauto.NewHistogramVec(prometheus.HistogramOpts{
 	Name: "nats_event_delivery_count",
 }, []string{"topic", "msg_id"})
 
+var PanicRecoveryCount = promauto.NewCounterVec(prometheus.CounterOpts{
+	Name: "Panic_Recovery_Count",
```
Comment (adi6859) [inline pubsub-lib/metrics/metrics.go:36]: move to constant

**Review by adi6859 — APPROVED** [review]

File: pubsub-lib/metrics/metrics.go [hunk 2]
```diff
@@ -1,36 +1,41 @@
...
-	Name: "nats_event_publish_time",
-}, []string{"topic"})
+	Name: constants.NATS_EVENT_PUBLISH_TIME,
+}, []string{constants.TOPIC})
 
 var NatsEventDeliveryCount = promauto.NewHistogramVec(prometheus.HistogramOpts{
-	Name: "nats_event_delivery_count",
-}, []string{"topic", "msg_id"})
+	Name: constants.NATS_EVENT_DELIVERY_COUNT,
+}, []string{constants.TOPIC, constants.MESSAGE_ID})
```
Comment (kripanshdevtron) [inline pubsub-lib/metrics/metrics.go:34]: need to remove message id as well.

File: middlewares/recovery.go [hunk 3]
```diff
@@ -14,6 +16,7 @@ func Recovery(next http.Handler) http.Handler {
 		defer func() {
 			err := recover()
 			if err != nil {
+				metrics.IncPanicRecoveryCount("handler", r.Host, r.Method, r.RequestURI, fmt.Sprint(err))
```
Comment (kripanshdevtron) [inline middlewares/recovery.go:19]: we cannot log error message as labels

**Review by gireesh-devtron — APPROVED** [review]
LGTM

**Review by gireesh-devtron — APPROVED** [review]
lgtm

**Review by kripanshdevtron — APPROVED** [review]

**Pull request merged**